### PR TITLE
docs: add "Writing Tests" example

### LINKED ISSUE #1222: proposal(docs): add example on how to test route handlers
`createHandler()` was implemented in #622. It'd be great to have an example in the new "Examples" section of the documentation of how testing one's Fresh project can be done with this new function.

### PR DESCRIPTION
**Pull request by Octo8080X**
Fixes [#1222](https://github.com/denoland/fresh/issues/1222)

Add example implementation and documentation of test code using `createHandler()` newly added by [#622](https://github.com/denoland/fresh/pull/622)

### CODE REVIEW

**Comment by iuioiua** [comment]
Hi, @Octo8080X. Apologies, I wasn't clear enough that we'd like this in the new examples section of the documentation. It'll only require snippets, which will mean less maintenance in the future. Can you rework this as a `/docs/examples/testing.md` (or similar) file?

Also, could you please use Deno's Standard Library for assertion functions? You can check it out [here](https://deno.land/std@0.190.0/testing/asserts.ts?doc=).

**Comment by Octo8080X** [comment]
Hi, @iuioiua san.

> Can you rework this as a /docs/examples/testing.md (or similar) file?

OK. `/docs/getting-started/create-test.md`, I'll let `/docs\examples\testing.md` remake it.

> Also, could you please use Deno's Standard Library for assertion functions?

I am sorry, but I used `assertEquals` that existed in `https://deno.land/std@0.190.0/testing/asserts.ts`. Was it insufficient?
If you don't want me to assume an import map, or if you mean that you don't want me to use superdeno and deno_dom, please let me know.

**Comment by iuioiua** [comment]
> OK. `/docs/getting-started/create-test.md`, I'll let `/docs\examples\testing.md` remake it.
> 
> >

👍🏾 

> If you don't want me to assume an import map, or if you mean that you don't want me to use superdeno and deno_dom, please let me know.

Yeah, I think just using assertions from `std` is good enough. I don't think superdeno and deno_dom are needed.

**Comment by Octo8080X** [comment]
@iuioiua san
Thanks for letting me know.

Removed dependencies on superdeno and deno_dom and moved documents.
Please check.

File: docs/examples/testing.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    "$fresh/": "../../",
+    "preact": "https://esm.sh/preact@10.13.1",
+    "preact/": "https://esm.sh/preact@10.13.1/",
+    "preact-render-to-string": "https://esm.sh/*preact-render-to-string@5.2.6",
+    "superdeno/": "https://deno.land/x/superdeno/",
+    "std/testing/": "https://deno.land/std@0.190.0/testing/",
+    "deno_dom/": "https://deno.land/x/deno_dom/"
+  }
+}
+```
```
Comment (iuioiua) [inline docs/examples/testing.md]: I think we can remove this.

File: docs/examples/testing.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+import type { ConnInfo } from "../../../src/server/deps.ts";
+
+const CONN_INFO: ConnInfo = {
+  localAddr: { hostname: "127.0.0.1", port: 8000, transport: "tcp" },
+  remoteAddr: { hostname: "127.0.0.1", port: 53496, transport: "tcp" },
+};
+
+Deno.test("HTTP assert test.", async (t) => {
+  await t.step("#1 GET /", async () => {
+    const handler = await createHandler(manifest);
```
Comment (iuioiua) [inline docs/examples/testing.md]: `handler` can be moved to the start of the test and only declared once, instead at every test step.

File: docs/examples/testing.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      Hello Foo!
+    </div>
+  );
+}
+```
+
+### Test code
+
+```ts
+// examples\test\test\main_test.ts
```
Comment (iuioiua) [inline docs/examples/testing.md]:
```suggestion
// tests/main_test.ts
```

**Review by iuioiua — CHANGES_REQUESTED** [review]
Can you please delete everything outside of the `docs/examples` folder?

A few other notes:
- Are you able to reduce some of the spacing between lines? It seems a bit too spread out.
- I think we can remove the last test step.
- Check out https://github.com/denoland/fresh/blob/main/tests/main_test.ts if you'd like some guidance.

**Comment by Octo8080X** [comment]
I missed the instruction to delete examples. My apologies.
I was able to delete it.
(Only docs/examples/testing.md was added.)

I would appreciate it if you could check my remarks to test code, as I have also reflected them.

**Review by iuioiua — APPROVED** [review]
I've made a few tweaks. Now, LGTM! Thanks, @Octo8080X.

Note: do not merge this until the next release is cut, as `createHandler()` is not yet part of the API.

**Comment by iuioiua** [comment]
On second thought, this might belong in the "Concepts" section instead of "Examples"...

**Review by lucacasonato — APPROVED** [review]
I think this being in the "Examples" section is ok. Thanks!

**Comment by iuioiua** [comment]
Ah, I should have followed my own advice and merged this after the next release was cut. Hopefully, this doesn't cause too much headache, as the new release is being published soon. Apologies. My bad.

**Pull request merged**